src/expr/src/scalar/func.rs

```diff
@@ -4210,6 +4214,33 @@ fn trim_whitespace<'a>(a: Datum<'a>) -> Datum<'a> {
     Datum::from(a.unwrap_str().trim_matches(' '))
 }
 
+fn right<'a>(a: Datum<'a>, b: Datum<'a>) -> Datum<'a> {
+    let string: &'a str = a.unwrap_str();
+    let n = b.unwrap_int64();
+
+    // the index of every char in bytes
+    let byte_indices = string.char_indices().map(|(i, _)| i);
```

Comment:
This chunk of code might be a bit clearer with direct index math instead of having to do the `rev().skip()` work. It would look something like:
```
    let start_index = if n >= 0 {
        if n >= string_len {
            0
        } else {
            string_len - n
        }
    } else {
        if n < 0 {
            string_len
        } else {
            n.abs()
        }
    };
```

Reply:
I find this one quite tricky. If I want to get `string_len` I need to do a `string.chars().count()`, which is `O(string_len)`.  And to then convert the `start_index` (in "char space") to a byte index I again have to traverse the iterator and sum up the bytes.

That's why I chose the somewhat complicated-looking reverse iteration. This way I only need to step `n` steps from the back, and I end up directly at the right byte index.

My solution isn't correct yet, as the test you suggested shows, but I could fix it. 😅

What do you think?

Reply:
I could be quite wrong on the string iteration/length business. Maybe I drank too much of the Rust-Iterators-are-cool Kool-Aid... 😅

Reply:
I pushed a change that makes it slightly less Iterator-y. I also added error handling with `usize::try_from()` but from looking at how the code is now, I don't think it can ever fail.

Reply:
You were totally right! It turns out [`.len()`](https://doc.rust-lang.org/std/string/struct.String.html#method.len) returns the length of the `String` in bytes. Since chars/graphemes can be larger than a single byte, the index math I was suggesting would be wrong in these cases.